woocommerce/woocommerce-ios
# [Issue Refund] Add Analytics

closes #2793

# Why 

To complete the "Issue Refund" feature we need to match the tracking that is already implemented in Android.
Specifically, the following events are added:

- refund_create
- refund_create_failed
- refund_create_success
- create_order_refund_select_all_items_button_tapped
- create_order_refund_shipping_option_tapped
- create_order_refund_item_quantity_dialog_opened
- create_order_refund_summary_refund_button_tapped
- create_order_refund_next_button_tapped
- create_order_refund_shipping_option_tapped 

# How
- Added track identifiers enum cases to `WooAnalyticsStat`
- Added track events methods to `WooAnalyticsEvents`
- Fire methods from `IssueRefundViewModel` and `RefundConfirmationViewModel`
- Add test cases

# Testing Steps
- Confirm the analytics are being tracked correctly and printed in the console while issuing a refund

**or**

- Trust the unit tests 😅 

Update release notes:
